src/Utilities/matrix_of_constraints.jl

```diff
@@ -80,8 +80,8 @@ end
         coefficients::AT
         constants::BT
         sets::ST
-        are_indices_mapped::BitSet
-        caches::Union{Nothing, Vector}
+        caches::Union{Nothing,Vector}
+        are_indices_mapped::Union{Nothing,Vector{BitSet}}
```

Comment:
What is the point of making these `Union{Nothing` if `empty!` sets them to empty vectors?

Reply:
If you add or remove set types to the `sets` field then the length of `caches` and `are_indices_mapped` should change so creating new ones at `empty!` is more robust to that.

Reply:
To clarify: do we ever use the fact that these are `nothing`? I can't see so. So the caches could be initialized as `Any[]` and `BitSet[]`?

Reply:
The `nothing` is what's modeling best the situation I feel. When you are not in `copy_to`, `caches` and `are_indices_mapped` don't make sense so it's appropriate that it's `nothing` so that it's explicit.

Reply:
But this assumes you call `empty!` at the start of `copy_to`? Did you notice any compilation issues with the union in the functions? The compiler might not like we iterate over a `Union{Nothing,Vector}`. The code is simpler if we make this `caches::Vector{Any}` and `::Vector{BitSet}`, and add some comments to the docstring.